### PR TITLE
GroupId check should be case insensitive

### LINKED ISSUE #13684: GroupId checks in DisplayDriver.UpdaetAsync are case sensitive, but URLs are not
### Describe the bug
If I go to a settings page in the Orchard Core admin panel that has a corresponding DisplayDriver, the URL of the page is not case sensitive, but the `if (context.GroupId == GroupId)` check in the override of DisplayDriver.UpdateAsync is case sensitive. This means that I can enter two different URLs for a settings page, both of which will work, but only one of which will actually cause any updates to be applied when the form is submitted. When I searched for that line of code across the Orchard Core 1.5.0 solution (which is what I'm currently running on), I find it in 16 drivers, so I would expect this issue affects all of them.

### To Reproduce
1. In the admin panel of any Orchard Core site, go to ~/Admin/Settings/general
2. Update the site name and hit Save. You should see your changes applied.
3. Change the URL to ~/Admin/Settings/**General**
4. The page should load correctly, but the hidden GroupId field is now set to "General" rather than "general"
5. Update the site name again and hit Save. You should see a message indicating that the settings have been updated, but they haven't actually been applied.

### Expected behavior
The GroupId comparison in DisplayDriver.UpdateAsync should be case insensitive. If the page loads correctly for a given URL, submitting the page should work as expected rather than mysteriously failing to update the settings.

### Screenshots
Link to GIF of described reproduction steps
https://i.imgur.com/Pd1lU27.gif

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #13684

### CODE REVIEW

**Comment by ryanwawr** [comment]
@hishamco Thank you!

File: src/OrchardCore.Modules/OrchardCore.Search/Drivers/SearchSettingsDisplayDriver.cs [hunk 1]
```diff
@@ -57,18 +57,21 @@ public override async Task<IDisplayResult> UpdateAsync(SearchSettings section, B
...
-            if (context.GroupId != GroupId || !await _authorizationService.AuthorizeAsync(user, Permissions.ManageSearchSettings))
+            if (!await _authorizationService.AuthorizeAsync(user, Permissions.ManageSearchSettings))
             {
                 return null;
             }
 
-            var model = new SearchSettingsViewModel();
-
-            if (await context.Updater.TryUpdateModelAsync(model, Prefix))
+            if (context.GroupId.Equals(GroupId, StringComparison.OrdinalIgnoreCase))
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Search/Drivers/SearchSettingsDisplayDriver.cs:65]: I would add his check back to line 60 to reduce the indentation.

File: src/OrchardCore.Modules/OrchardCore.ContentLocalization/Drivers/ContentCulturePickerSettingsDriver.cs [hunk 2]
```diff
@@ -48,7 +49,7 @@ public override async Task<IDisplayResult> UpdateAsync(ContentCulturePickerSetti
                 return null;
             }
 
-            if (context.GroupId == GroupId)
+            if (context.GroupId.Equals(GroupId, StringComparison.OrdinalIgnoreCase))
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.ContentLocalization/Drivers/ContentCulturePickerSettingsDriver.cs:52]: you may want to create an extension to the `BuildShapeContext` class that can be reused when comparing group?

something like this

```
public static class BuildShapeContextExtensions
{
	public static class SameGroupAs(this BuildShapeContext context, string groupId)
	{
		return context.GroupId.Equals(groupId, StringComparison.OrdinalIgnoreCase);
	}
}
```

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.ContentLocalization/Drivers/ContentCulturePickerSettingsDriver.cs:52]: I don't see any value here to create an extension with a single line code, but let me revise the PR again and see what we could come up with

File: src/OrchardCore.Modules/OrchardCore.Search/Drivers/SearchSettingsDisplayDriver.cs (hunk 1 above)
Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Search/Drivers/SearchSettingsDisplayDriver.cs:65]: We did the same check after permission check in all the places

**Pull request merged**